config/CR01R02v4/CMakeBuild.config

```diff
@@ -0,0 +1,148 @@
+########################################################################################################################
+# Configuration options, used by build systems and scripts.
+########################################################################################################################
+#
+# The configuration settings that are used for releases are set in:
+#    source/conf/cmake/CMakeBuild.config.default
+#
+# The default settings here are for development. The settings of CMakeBuild.config.default will be overwritten.
+#
+# If you want other settings on your system, do not just this file (it will be commited to the code repository).
+#
+# Recommended:
+#
+#   + Add the custom settings to a new file in this directory: config/default/CMakeBuild.overwrite.config
+#     + If those are runtime settings, you can also use: config/default/CMakeBuild.runtime.config
+#     + Run cmake and make with the default target.
+#   + Add a new directory in config, say config/feature
+#     + Copy this file to that directory and change whatever you like.
+#     + No need to commit this new directory to git, feel free again to use overwrite and runtime config.
+#     + Run cmake and make with the new target (here called feature).
+#
+
+# The name that is advertised for this target. Should be 4 characters or less.
+BLUETOOTH_NAME=CR40
+
+# By default, a Crownstone has its serial disabled, which can be enabled via a bluetooth command.
+# You can change the default value:
+CS_SERIAL_ENABLED=SERIAL_ENABLE_RX_AND_TX
+
+# Set the log level.
+# A complete list of options can be found in CMakeBuild.config.default
+# Always check if you can compile with maximum verbosity: SERIAL_DEBUG.
+SERIAL_VERBOSITY=SERIAL_INFO
+
+# Enable the binary protocol if you want to use a UART library.
+# Disable the binary protocol if you want to use a serial communication program, like minicom.
+CS_UART_BINARY_PROTOCOL_ENABLED=1
+
+# Enable microapps
+BUILD_MICROAPP_SUPPORT=1
+
+# Automatically enable microapp on boot
+AUTO_ENABLE_MICROAPP_ON_BOOT=1
+
+# Enable gpio
+BUILD_GPIOTE=1
+
+GPIO_PIN1_INDEX=0
+GPIO_PIN2_INDEX=1
+GPIO_PIN3_INDEX=2
+GPIO_PIN4_INDEX=3
+
+BUTTON1_INDEX=0
+BUTTON2_INDEX=1
+BUTTON3_INDEX=2
+BUTTON4_INDEX=3
+
+LED1_INDEX=0
+LED2_INDEX=1
+LED3_INDEX=2
+LED4_INDEX=3
+
+########################################################################################################################
+# Settings for the nRF52840
+# See CMakeBuild.config.default for explanation.
+########################################################################################################################
+
+# The hardware board where you want to upload the firmware
+# For the complete overview of the hardware boards, see CMakeBuild.config.default
+HARDWARE_BOARD=CR01R02v4
+
+# The hardware board as defined within the Nordic code (e.g. for internal pin layout).
+NORDIC_HARDWARE_BOARD=CUSTOM
```

Comment:
What does this do?

Reply:
This is a precompiler definition `NORDIC_HARDWARE_BOARD` which is used in the Nordic code. In our cmake files it is used in `./source/conf/cmake/crownstone.defs.cmake`

```
IF(DEFINED NORDIC_HARDWARE_BOARD)
    ADD_DEFINITIONS("-DBOARD_${NORDIC_HARDWARE_BOARD}")
ENDIF()
```

At certain spots in the Nordic code there are macros that check with board is in use. If no `DBOARD_` macro has been defined it picks a board at random or the compilation fails. A grep on `BOARD_CUSTOM` would have been sufficient to answer your own question, see `./components/boards/boards.h`. If you would like to see a more detailed explanation on macros like these, feel free to expand.

Reply:
It would be nice to add this option in the explanation of this field in the `CMakeBuild.config.default`.

Reply:
I think if it's simple things like adding a few comments, the reviewer can just do so?